test/e2e/e2e_test.go

```diff
@@ -1106,35 +1107,18 @@ var _ = Describe("e2e non-vxlan external gateway and update validation", func()
 		}
 		// Create the pod that will be used as the source for the connectivity test
 		createGenericPod(f, srcPingPodName, ciWorkerNodeSrc, command)
-
-		// Wait for pod exgw setup to be almost ready
-		wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
-			jsonFlag := "jsonpath='{.metadata.annotations.k8s\\.ovn\\.org/routing-external-gws}'"
-			kubectlOut, err := framework.RunKubectl("get", "pod", ovnNsFlag, srcPingPodName, jsonFlag)
-			if err != nil {
+		// wait for pod setup to return a valid address
+		wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
```

Comment:
wait.PollImmediate return an error if the function does not return true during the retryTimeout, we should check it or this is just the same as `time.Sleep( retryTimeout)` 

```go
err = wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
...
}
if err != nil {
framework.Failf("Error trying to get the pod IP address")
}
```

Reply:
also, to expand on the function usage as it seems the upstreadm docs are not clear,  if the  function checking the conditions returns an error, the errors is returned directly and no additional checks are done

Reply:
Ty, good catch. Went back and changed the other instances to check for an error returned as well.